second commit

DmitryChebokhin opened the pull request.

DmitryChebokhin committed:
second commit

pomegranate282922 commented on src/main/java/BaseApi.java:
```diff
@@ -0,0 +1,15 @@
+
+public class BaseApi {
+
+    public static RequestSpecification getBaseSpecification() {
```
Отлично что используется req spec

pomegranate282922 commented on src/main/java/UserSetGet.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    public static  UserSetGet generateUser() {
+        String email = RandomStringUtils.randomAlphabetic(3,12) + "@gmail.com";
```
Отлично что есть рандомизация

pomegranate282922 commented on src/test/java/OrderCreateTest.java:
```diff
@@ -0,0 +1,86 @@
+
+public class OrderCreateTest {
+
+    String correctIngredients = "{\n\"ingredients\": [\"61c0c5a71d1f82001bdaaa6d\",\"61c0c5a71d1f82001bdaaa6f\"]\n}";
```
Валидные ингредиенты не хардкодим, а получаем из соответсвующей ручки 
Нужно исправить

pomegranate282922 commented on src/test/java/OrderCreateTest.java:
```diff
@@ -0,0 +1,86 @@
+
+    @Test
+    @DisplayName("Создание заказа без авторизации и без списка ингредиентов")
+    public void CreateOrderWithoutAuthAndWithoutIngredientsTest() {
```
Методы именуем camelCase 
Нужно исправить

pomegranate282922 commented on src/test/java/UserLoginTest.java:
```diff
@@ -0,0 +1,71 @@
+
+    @After
+    public void cleanUp() {
+        if (authToken != null) {
```
Не везде удаляется юзер 
Нужно исправить

pomegranate282922 replied:
Не исправлено

pomegranate282922 reviewed (commented):
Работа хорошая! Все тесты реализованы. Используется рандомизация. Всем тестам и шагам даны наименования.
Замечания:
1. Не везде удаляется созданный юзер 
2. Нейминг методов 
3. Убрать хардкод валидных ингредиентов

DmitryChebokhin committed:
new commit

Исправление замечаний

pomegranate282922 commented on src/test/java/OrderCreateTest.java:
```diff
@@ -0,0 +1,102 @@
+
+    @Test
+    @DisplayName("Создание заказа с авторизацией и списком ингредиентов")
+    public void CreateOrderWithAuthAndWithIngredientsTest() {
```
Нейминг методов camelCase 
Нужно исправить

pomegranate282922 commented on src/test/java/OrderGetTest.java:
```diff
@@ -0,0 +1,54 @@
+    OrderApi orderClient;
+    UserApi userClient;
+    String authToken;
+    String correctIngredients = "{\n\"ingredients\": [\"61c0c5a71d1f82001bdaaa75\",\"61c0c5a71d1f82001bdaaa70\"]\n}";
```
А здесь?
Нужно исправить

pomegranate282922 commented on src/test/java/UserCreateTest.java:
```diff
@@ -0,0 +1,80 @@
+
+    @After
+    public void cleanUp() {
+        if (authToken != null) {
```
Не везде удаляется юзер 
Нужно исправить

pomegranate282922 reviewed (commented):
Привет! Кое-что исправил, но не все( 
1. Нейминг проверить по всем коде 
2. Удаление юзера нужно сделать во всех тестах 
3. Убрать хардкод валидных ингредиентов отовсюду

DmitryChebokhin committed:
corrections

исправил нейминг, поправил удаление пользователя, убрал хардкод ингредиентов в OrderGetTest

pomegranate282922 commented on src/test/java/UserCreateTest.java:
```diff
@@ -23,58 +24,66 @@ public void setUp() {
     public void cleanUp() {
-        if (authToken != null) {
-            userClient.deleteUser(authToken.substring(7));
+        for (String authToken : authTokens) {
```
Список не нужен, метод этот вызывается после каждого теста. Но токен нужно везде заполнить

pomegranate282922 reviewed (approved):
Зачет!
